plugins/server/metrics/config_test.go

```diff
@@ -0,0 +1,96 @@
+package metrics
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestConfigValidation(t *testing.T) {
+	tests := []struct {
+		input   string
+		wantErr bool
+	}{
+		{
+			input:   `{}`,
+			wantErr: false,
+		},
+		{
+			input:   `{http_requests: {}}`,
+			wantErr: false,
+		},
+		{
+			input:   `{"http_requests": {"buckets": "not-a-array"}}`,
+			wantErr: true,
+		},
+		{
+			input:   `{"http_requests": {buckets": [1]}}`,
+			wantErr: false,
+		},
+		{
+			input:   `{"http_requests":{"buckets": "1"}}`,
+			wantErr: true,
+		},
+		{
+			input:   `{"http_requests":{"random_key": 0}}`,
+			wantErr: false,
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("TestConfigValidation_case_%d", i), func(t *testing.T) {
+			_, err := NewConfigBuilder().WithBytes([]byte(test.input)).Parse()
+			if err != nil && !test.wantErr {
+				t.Fail()
+			}
+			if err == nil && test.wantErr {
+				t.Fail()
+			}
+		})
+	}
+}
+
+func TestConfigValue(t *testing.T) {
+	tests := []struct {
+		input         string
+		expectedValue []float64
+	}{
+		{
+			input:         `{}`,
+			expectedValue: defaultHTTPRequestBuckets,
+		},
+		{
+			input:         `{"http_requests":{}}`,
+			expectedValue: defaultHTTPRequestBuckets,
+		},
+		{
+			input:         `{"http_requests":{"buckets":[0.1, 0.2, 0.3, 4]}}`,
+			expectedValue: []float64{0.1, 0.2, 0.3, 4},
+		},
```

Comment:
Probably a test case to verify the array element type would be useful.

Reply:
added a test but in the other method.
if the values are off (strings, combination of strings with floats, etc) the checks will throw exceptions